sdk/examples/intkey_cxx/int_key_transaction_processor.cpp

```diff
@@ -0,0 +1,292 @@
+/* Copyright 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------*/
+#include <iostream>
+#include <string>
+
+#include "log4cxx/logger.h"
+#include "log4cxx/basicconfigurator.h"
+
+#include "sawtooth/address_mapper.h"
+#include "sawtooth/transaction_processor.h"
+#include "json.hpp"
+
+using namespace log4cxx;
+
+using namespace nlohmann;
+
+static log4cxx::LoggerPtr logger(log4cxx::Logger::getLogger
+    ("sawtooth.IntKeyCXX"));
+
+static const std::string INTKEY_NAMESPACE = "intkey";
+
+// Custom Exception type defined to indicate any
+// failure in process a transaction.
+class InvalidTransaction: public std::runtime_error {
+ public:
+    explicit InvalidTransaction(std::string const& error)
+        : std::runtime_error(error)
+    {}
+};
+
+
+// utility function to provide copy conversion from vector of bytes
+// to a stl string container.
+void ToString(std::string* out, const std::vector<std::uint8_t>& in) {
+    out->resize(in.size());
+    const char* data = reinterpret_cast<const char*>(&in[0]);
+    out->assign(data, data+in.size());
+}
+
+// utility function to provide copy conversion from stl string container
+// to a vector of bytes.
+void ToVector(std::vector<std::uint8_t>* out, const std::string& in) {
+    out->resize(in.length());
+    memcpy(&((*out)[0]), in.data(), in.length());
+}
+
+// Handles the processing of a IntKey transactions.
+class IntKeyApplicator:  public sawtooth::TransactionApplicator {
+ public:
+    IntKeyApplicator(sawtooth::TransactionUPtr txn,
+        sawtooth::GlobalStateUPtr state) :
+        TransactionApplicator(std::move(txn), std::move(state)),
+            address_mapper(INTKEY_NAMESPACE) { }
+
+    void Apply() {
+        LOG4CXX_DEBUG(logger, "IntKeyApplicator::Apply");
+
+        const std::string& raw_data = this->txn->payload();
+        std::vector<uint8_t> data_vector(raw_data.begin(), raw_data.end());
+        json intkey_cmd = json::from_cbor(data_vector);
+
+        if (!intkey_cmd.is_object()) {
+            throw InvalidTransaction("CBOR Object as the encoded command");
+        }
+        auto verb_it = intkey_cmd.find("Verb");
+        if (verb_it == intkey_cmd.end()) {
```

Comment:
A nit, but I would tend to write:

`if (intkey_cmd.end() == verb_it)` to prevent accidental assignment, although g++ is probably much better at flagging that warning that MSFT compilers of the past.